lib/ledger_sync/adaptors/ledger_serializer_attribute.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require_relative 'ledger_serializer_type/value'
+Gem.find_files('ledger_sync/adaptors/ledger_serializer_type/**/*.rb').each { |path| require path }
+
+module LedgerSync
+  module Adaptors
+    class LedgerSerializerAttribute
+      attr_reader :block,
+                  :id,
+                  :ledger_attribute,
+                  :resource_attribute,
+                  :resource_class,
+                  :serializer,
+                  :type
+
+      def initialize(block: nil, id: false, ledger_attribute:, resource_attribute: nil, resource_class: nil, serializer: nil, type: LedgerSerializerType::Value)
```

Comment:
I prefer keywords as they enforce that things are passed correctly.  Otherwise you have to do validation on the hash, keys, and values (which is more code).  It is a lot thought.  I'm guessing there is probably a better way to model this, by splitting up some responsibilities.  I just kept having to add things to make types work downstream.